plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStore.java

```diff
@@ -314,17 +314,39 @@ void deleteBlobs(Collection<String> blobNames) throws IOException {
             return;
         }
         final List<BlobId> blobIdsToDelete = blobNames.stream().map(blob -> BlobId.of(bucketName, blob)).collect(Collectors.toList());
-        final List<Boolean> deletedStatuses = SocketAccess.doPrivilegedIOException(() -> client().delete(blobIdsToDelete));
+        final List<Boolean> deletedStatuses = SocketAccess.doPrivilegedIOException(() -> {
+            final StorageBatch batch = client().batch();
+            final List<Boolean> results = Lists.newArrayList();
+            for (BlobId blob : blobIdsToDelete) {
+                batch.delete(blob).notify(
+                    new BatchResult.Callback<>() {
+                        @Override
+                        public void success(Boolean result) {
+                            results.add(result);
+                        }
+
+                        @Override
+                        public void error(StorageException exception) {
+                            if (exception.getCode() == 404) {
+                                results.add(Boolean.TRUE);
+                            } else {
+                                results.add(Boolean.FALSE);
+                            }
+                        }
+                    });
+            }
+            batch.submit();
+            return Collections.unmodifiableList(results);
+        });
         assert blobIdsToDelete.size() == deletedStatuses.size();
         boolean failed = false;
         for (int i = 0; i < blobIdsToDelete.size(); i++) {
             if (deletedStatuses.get(i) == false) {
-                logger.error("Failed to delete blob [{}] in bucket [{}]", blobIdsToDelete.get(i).getName(), bucketName);
                 failed = true;
             }
         }
         if (failed) {
-            throw new IOException("Failed to delete all [" + blobIdsToDelete.size() + "] blobs");
+            throw new IOException("Failed to delete the following blobs " + blobIdsToDelete);
```

Comment:
perhaps just `"Failed to delete blobs " + blobIdsToDelete`

Reply:
done